backend/controllers/order.controller.js

```diff
@@ -54,9 +63,45 @@ const getOrderById = asyncHandler(async (req, res) => {
     }
   } catch (error) {
     console.log(error);
-    res.status(404);
-    throw new Error("Order not found");
+    res.status(error.code);
+    throw new Error(error.message);
   }
 });
 
-export { createOrder, getOrderById };
+/**
+ * @desc        Update Order Status
+ * @route       PATCH /api/orders/:orderId/status
+ * @access      Private
+ */
+
+const updateOrderStatus = asyncHandler(async (req, res) => {
+  try {
+    const {orderId} = req.params
+    const {status} = req.query
+    const user = await User.findById(req.user._id);
+    if(user){
+      const order = await Order.findById(orderId).populate('user')
+      if(order && order.user.email ===  user.email){
```

Comment:
Also, when updating an order, only the user assigned to that order is able to update it. @giangnguyen13 @hanguy2806